synapse/logging/opentracing.py

```diff
@@ -855,12 +855,12 @@ def err_back(result: R) -> R:
                     scope.__exit__(type(e), None, e.__traceback__)
                     raise
 
-        return _trace_inner
+        return _trace_inner  # type: ignore[return-value]
```

Comment:
What's the error here, out of interest?

Reply:
```
synapse/logging/opentracing.py:858: error: Incompatible return value type (got "Callable[P, Coroutine[Any, Any, R]]", expected "Callable[P, R]")  [return-value]
```

Actually now that I split this up I might be able to do simple overloads with returning `Awaitable[R]` in a couple situations and remove this ignore?

Reply:
I'm not sure that's possible actually. Any idea if we should do something about this one?

Reply:
As sad as I am to leave a typing problem lying around like this, I think I'd probably leave the ignore as is. The important thing is that we accurately annotate the decorator itself; its internals don't have an effect on the rest of the function.

Reply:
> its internals don't have an effect on the rest of the function.

This was kind of my conclusion after messing with it for a bit.